hudi-client/hudi-client-common/src/test/java/org/apache/hudi/client/transaction/TestLockManager.java

```diff
@@ -0,0 +1,119 @@
+import org.apache.hudi.common.model.HoodieFailedWritesCleaningPolicy;
+import org.apache.hudi.common.model.WriteConcurrencyMode;
+import org.apache.hudi.common.testutils.HoodieCommonTestHarness;
+import org.apache.hudi.config.HoodieCleanConfig;
+import org.apache.hudi.config.HoodieLockConfig;
+import org.apache.hudi.config.HoodieWriteConfig;
+
+import org.apache.curator.test.TestingServer;
+import org.apache.log4j.LogManager;
+import org.apache.log4j.Logger;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.io.IOException;
+import java.lang.reflect.Field;
+
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertNull;
+
+public class TestLockManager extends HoodieCommonTestHarness {
+
+  private static final Logger LOG = LogManager.getLogger(TestLockManager.class);
+
+  private static TestingServer server;
+  private static String zk_basePath = "/hudi/test/lock";
+  private static String key = "table1";
+
+  HoodieWriteConfig writeConfig;
+  LockManager lockManager;
+
+  @BeforeAll
+  public static void setup() {
+    while (server == null) {
+      try {
+        server = new TestingServer();
+      } catch (Exception e) {
+        LOG.error("Getting bind exception - retrying to allocate server");
+        server = null;
+      }
+    }
+  }
+
+  @AfterAll
+  public static void tearDown() throws IOException {
+    if (server != null) {
+      server.close();
+    }
+  }
+
+  private HoodieWriteConfig getWriteConfig() {
+    return HoodieWriteConfig.newBuilder()
+        .withPath(basePath)
+        .withCleanConfig(HoodieCleanConfig.newBuilder()
+            .withFailedWritesCleaningPolicy(HoodieFailedWritesCleaningPolicy.LAZY)
+            .build())
+        .withWriteConcurrencyMode(WriteConcurrencyMode.OPTIMISTIC_CONCURRENCY_CONTROL)
+        .withLockConfig(HoodieLockConfig.newBuilder()
+            .withLockProvider(ZookeeperBasedLockProvider.class)
+            .withZkBasePath(zk_basePath)
+            .withZkLockKey(key)
+            .withZkQuorum(server.getConnectString())
+            .build())
+        .build();
+  }
+
+  @BeforeEach
+  private void init() throws IOException {
+    initPath();
+    initMetaClient();
+    this.writeConfig = getWriteConfig();
+    this.lockManager = new LockManager(this.writeConfig, this.metaClient.getFs());
+  }
+
+  @Test
+  public void testLockAndUnlock() throws NoSuchFieldException, IllegalAccessException{
+
+    Field lockProvider = lockManager.getClass().getDeclaredField("lockProvider");
+    lockProvider.setAccessible(true);
+
+    assertDoesNotThrow(() -> {
+      lockManager.lock();
+    });
+
+    assertNotNull(lockProvider.get(lockManager));
+
+    assertDoesNotThrow(() -> {
+      lockManager.unlock();
```

Comment:
We can use the mokito to check the `#close` method invoke instead, take `TestCloudWatchMetricsReporter` for an example.

Reply:
+1